src/DataFrameIndexingMsg.chpl

```diff
@@ -0,0 +1,204 @@
+                }
+                when ("Strings") {
+                    dfiLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),"Element at %i is Strings".format(i));
+                    var args: [1..2] string = [ele_parts[2], iname];
+                    var repTup = segPdarrayIndex("str", args, st);
+                    if repTup.msgType == MsgType.ERROR {
+                        throw new IllegalArgumentError(repTup.msg);
+                    }
+
+                    rpm = "%jt".format("Strings+%s+%s".format(col_name, repTup.msg));
+                }
+                when ("pdarray"){
+                    dfiLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),"Element at %i is pdarray".format(i));
+                    var gCol: borrowed GenSymEntry = getGenericTypedArrayEntry(ele_parts[2], st);
+                    select (gCol.dtype) {
+                        when (DType.Int64) {
+                            var col_vals = toSymEntry(gCol, int);
+                            rpm = "%jt".format(dfIdxHelper(idx, col_vals, st, col_name));
+                        }
+                        when (DType.UInt64) {
+                            var col_vals = toSymEntry(gCol, uint);
+                            rpm = "%jt".format(dfIdxHelper(idx, col_vals, st, col_name));
+                        }
+                        when (DType.Bool) {
+                            var col_vals = toSymEntry(gCol, bool);
+                            rpm = "%jt".format(dfIdxHelper(idx, col_vals, st, col_name));
+                        }
+                        when (DType.Float64){
+                            var col_vals = toSymEntry(gCol, real);
+                            rpm = "%jt".format(dfIdxHelper(idx, col_vals, st, col_name));
+                        }
+                        otherwise {
+                            var errorMsg = notImplementedError(pn,dtype2str(gCol.dtype));
+                            dfiLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                            throw new IllegalArgumentError(errorMsg);
+                        }
+                    }
+                }
+                when ("SegArray"){
+                    dfiLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),"Element at %i is SegArray".format(i));
+                    ref segments_name = ele_parts[2];
+                    ref values_name = ele_parts[3];
+                    dfiLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),"Segments Name: %s, Values Name: %s".format(segments_name, values_name));
+
+                    var gSeg: borrowed GenSymEntry = getGenericTypedArrayEntry(segments_name, st);
+                    var segments = toSymEntry(gSeg, int);
+                    var gVal: borrowed GenSymEntry = getGenericTypedArrayEntry(values_name, st);
+                    var values = toSymEntry(gVal, int);
```

Comment:
does segarray also support `uint` as possible type for values?

Reply:
updated to properly handle supported types.